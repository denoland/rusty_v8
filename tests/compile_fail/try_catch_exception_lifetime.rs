// Copyright 2019-2020 the Deno authors. All rights reserved. MIT license.

pub fn main() {
  let mut isolate = v8::Isolate::new(mock());
  v8::make_handle_scope!(scope1, &mut isolate);
  let context = v8::Context::new(scope1, Default::default());
  let mut scope = v8::ContextScope::new(scope1, context);

  let _exception = {
    v8::make_handle_scope!(scope3, &mut *scope);
    v8::make_handle_scope!(scope4, scope3);
    let try_catch = std::pin::pin!(v8::TryCatch::new(&mut *scope4));
    let try_catch = try_catch.init();
    try_catch.exception().unwrap()
  };
}

fn mock<T>() -> T {
  unimplemented!()
}
