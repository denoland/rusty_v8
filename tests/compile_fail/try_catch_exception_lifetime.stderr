error[E0597]: `scope4` does not live long enough
  --> tests/compile_fail/try_catch_exception_lifetime.rs:11:5
   |
9  |   let _exception = {
   |       ---------- borrow later stored here
10 |     v8::make_handle_scope!(scope3, &mut scope);
11 |     v8::make_handle_scope!(scope4, scope3);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |     |
   |     borrowed value does not live long enough
   |     binding `scope4` declared here
...
15 |   };
   |   - `scope4` dropped here while still borrowed
   |
   = note: this error originates in the macro `v8::make_handle_scope` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0597]: `scope3` does not live long enough
  --> tests/compile_fail/try_catch_exception_lifetime.rs:10:5
   |
9  |   let _exception = {
   |       ---------- borrow later stored here
10 |     v8::make_handle_scope!(scope3, &mut scope);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |     |
   |     borrowed value does not live long enough
   |     binding `scope3` declared here
...
15 |   };
   |   - `scope3` dropped here while still borrowed
   |
   = note: this error originates in the macro `v8::make_handle_scope` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0597]: `scope3` does not live long enough
  --> tests/compile_fail/try_catch_exception_lifetime.rs:10:5
   |
9  |   let _exception = {
   |       ---------- borrow later stored here
10 |     v8::make_handle_scope!(scope3, &mut scope);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |     |
   |     borrowed value does not live long enough
   |     binding `scope3` declared here
...
15 |   };
   |   - `scope3` dropped here while still borrowed
   |
   = note: this error originates in the macro `v8::make_handle_scope` (in Nightly builds, run with -Z macro-backtrace for more info)
